root/entity/NotFound.js

```diff
@@ -0,0 +1,163 @@
+     You may wish to try and {search_url|search for it} instead.`,
+  ),
+  'isrc': N_l(
+    `This ISRC is not associated with any recordings. If you wish to associate
+     it with a recording, please {search_url|search for the recording}
+     and add it.`,
+  ),
+  'iswc': N_l(
+    `This ISWC is not associated with any works. If you wish to associate it
+     with a work, please {search_url|search for the work} and add it.`,
+  ),
+  'label': N_l(
+    `Sorry, we could not find a label with that MusicBrainz ID. 
+     You may wish to try and {search_url|search for it} instead.`,
+  ),
+  'otherlookup': N_l(
+    `Sorry, we could not find a MusicBrainz entity with that ID. 
+     You may wish to try and {search_url|search for it} instead.`,
+  ),
+  'place': N_l(
+    `Sorry, we could not find a place with that MusicBrainz ID. 
+     You may wish to try and {search_url|search for it} instead.`,
+  ),
+  'recording': N_l(
+    `Sorry, we could not find a recording with that MusicBrainz ID. 
+     You may wish to try and {search_url|search for it} instead.`,
+  ),
+  'relationship/linkattributetype': N_l(
+    `Sorry, we could not find a relationship attribute
+     with that MusicBrainz ID.`,
+  ),
+  'release': N_l(
+    `Sorry, we could not find a release with that MusicBrainz ID. 
+     You may wish to try and {search_url|search for it} instead.`,
+  ),
+  'release_group': N_l(
+    `Sorry, we could not find a release group with that MusicBrainz ID. 
+     You may wish to try and {search_url|search for it} instead.`,
+  ),
+  'series': N_l(
+    `Sorry, we could not find a series with that MusicBrainz ID. 
+     You may wish to try and {search_url|search for it} instead.`,
+  ),
+  'track': N_l(
+    `Sorry, we could not find neither a recording nor a track with that 
+     MusicBrainz ID.
+     You may wish to try and {search_url|search for it} instead.`,
+  ),
+  'url': N_l('Sorry, we could not find a URL with that MusicBrainz ID.'),
+  'user': N_l(
+    `Sorry, we could not find an editor with that name. 
+     You may wish to try and {search_url|search for them} instead.`,
+  ),
+  'work': N_l(
+    `Sorry, we could not find a work with that MusicBrainz ID. 
+     You may wish to try and {search_url|search for it} instead.`,
+  ),
+};
+
+const namespacesWithoutSearch = [
+  'cdtoc',
+  'collection',
+  'election',
+  'relationship/linkattributetype',
+  'url',
+];
+
+const NotFound = ({namespace}: {namespace: string}) => (
+  <NotFoundComponent title={notFoundTitlePicker[namespace]()}>
+    <p>
+      {namespacesWithoutSearch.includes(namespace) ? (
+        notFoundStringPicker[namespace]()
+      ) : (
+        expand2react(
+          notFoundStringPicker[namespace](),
+          namespace === 'genre' ? (
+            {
+              genre_list: '/genres',
+            }
+          ) : (
+            {
+              search_url: namespace === 'edit' ? '/search/edits' : '/search',
```

Comment:
I like the idea of this PR, but might do things slightly differently. I think one of the goals should be making adding a new not-found page as easy as possible (as simple as adding a new configuration block above), not necessarily making the code as compact as possible (though reducing the amount of code is nice too).

Adding a new page is not so easy now though, since you have to add some configuration above to two objects, then possibly `namespacesWithoutSearch`, and also possibly tweak the mess of ternary conditions down here if you somehow create a new exception.

My idea would be to have a single object like:

```JS
const notFoundPages = {
  'namespace': {
    title: '',
    message: '',
    args: {},     // expand2react args
    footer: null, // for e.g. elections
  },
};
```
and then change `NotFound` to simply look up the namespace in this object and display the relevant properties. So to add a new page, you'd only ever need to update the `notFoundPages` object by adding a single entry.

To help with any repetition that causes, we could have constants for args like `{search_url: '/search'}` which are shared among the objects that need it.

Reply:
That actually sounds much clearer than my version. Made the changes.